### PR TITLE
add 1.29 to kueue presubmits

### PR DESCRIPTION
**Pull request by kannon92**
Add 1.29 to Kueue e2e tests.

Question to reviewer:

Should I include these for all releases of Kueue and the release-blocking tests?

### CODE REVIEW

**Review by mimowo — COMMENTED** [review]
> Should I include these for all releases of Kueue and the release-blocking tests?

I would say yes, this is what we did for 1.28: https://github.com/kubernetes/test-infra/commit/d88a931a291c8cc39016c74702777d134639a05f

**Review by tenzen-y — COMMENTED** [review]
Also, I think that we can remove jobs for k8s v1.25 since that version has been already deprecated.

cc: @alculquicondor

**Comment by mimowo** [comment]
I think we should also add 1.29 here: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml

**Comment by tenzen-y** [comment]
> https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml

Yes, that's right. Thanks.

**Comment by alculquicondor** [comment]
I think we shouldn't remove 1.25 from release-0.5, but it should be fine to remove from main.

**Comment by kannon92** [comment]
So I will add 1.29 to main presubmit, add 1.29 to release-0.5, and remove 1.25 from main.

I noticed that release-0.3 does not go up to 1.28 so should I only add 1.29 to the latest release?

**Comment by alculquicondor** [comment]
We should probably remove tests for 0.3

We are not backporting that far anymore.

**Comment by kannon92** [comment]
I opened up a PR to drop release-0.3 jobs.

And this one should be ready for review.

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml
Comment (tenzen-y) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml:1]: @kannon92 Can we remove jobs for v1.25?

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml
Comment (tenzen-y) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:1]: ditto.

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml
Comment (kannon92) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml:1]: https://github.com/kubernetes/test-infra/pull/31481#issuecomment-1862958538

I updated this but forgot @alculquicondor comment that we shouldn't remove it for release 0.5.

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml
Comment (kannon92) [reply config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:1]: @alculquicondor wdyt about removing 1.25 from these tests?

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml
Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml:1]: Ah, @alculquicondor is correct. 
@kannon92 Can we keep the v1.25 jobs in release-0.5 and release-blocking?

Sorry for the confusion.

**Review by tenzen-y — COMMENTED** [review]
/approve

**Pull request merged**